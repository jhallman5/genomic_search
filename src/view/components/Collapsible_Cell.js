import React from 'react'
import { TableCell, withStyles, Icon } from '@material-ui/core'
import { PlayArrowOutlined } from '@material-ui/icons'

export class CollapsibleCell extends React.Component {
  constructor(props){
    super(props)
    this.state = {
      expanded: false
    }
    this.toggleExpand = this.toggleExpand.bind(this)
  }

 toggleExpand() {
   this.setState({ expanded: !this.state.expanded })
 }

  render(){
    return (
      <TableCell className={this.props.classes.cell}>
          <Icon>
            <PlayArrowOutlined
              className={
                !this.state.expanded
                ? this.props.classes.icon
                : this.props.classes.iconActive
              }
              onClick={this.toggleExpand}
            />
          </Icon>
      </TableCell>
    )
  }
}

const styles = theme => ({
  cell: {
    color: 'white',
    fontSize: 20
  },
  icon: {
    color: '#29736A',
  },
  iconActive: {
    color: '#3e5e9c',
    transform: 'rotate(90deg)'
  }
})


// {
//   this.props.gene.other_mappings &&

export default withStyles(styles)(CollapsibleCell)
