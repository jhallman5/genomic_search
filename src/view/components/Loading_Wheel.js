import React from 'react'
import { withStyles, CircularProgress } from '@material-ui/core/'

class LoadingWheel extends React.Component {
  constructor() {
    super()
    this.state = {
      completed: 0,
    }
    this.progress = this.progress.bind(this)
  }

  componentDidMount() {
    this.timer = setInterval(this.progress, 15)
  }

  componentWillUnmount() {
    clearInterval(this.timer)
  }

  progress(){
    const { completed } = this.state;
    this.setState({ completed: completed >= 100 ? 0 : completed + 1 })
  };

  render() {
    const { classes } = this.props;
    return (
      <div>
        <CircularProgress
          className={classes.progress}
          variant="determinate"
          size={75}
          value={this.state.completed}
        />
      </div>
    );
  }
}

const styles = theme => ({
  progress: {
    color: '#39B7AA',
    position: 'fixed',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)'
    },
})

export default withStyles(styles)(LoadingWheel)
