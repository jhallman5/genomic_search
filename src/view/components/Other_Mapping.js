import React from 'react'
import { withStyles } from '@material-ui/core'

const Other_Mapping = props => (
  <div className={props.classes.div}>
    {props.info}
  </div>
)

const styles = theme => ({
  div: {
    color: 'black',
    fontSize: 16
  }
})

export default withStyles(styles)(Other_Mapping)
