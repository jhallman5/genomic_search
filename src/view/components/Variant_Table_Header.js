import React from 'react'
import { Table, TableHead, TableRow, TableCell, withStyles } from '@material-ui/core'

const VariantTableHeader = props => (
  <TableHead>
    <TableRow className={props.classes.row}>
      <TableCell className={props.classes.cell}>Gene</TableCell>
      <TableCell className={props.classes.cell}>Nucleotide Change</TableCell>
      <TableCell className={props.classes.cell}>Protein Change</TableCell>
      <TableCell className={props.classes.cell}>Alias</TableCell>
      <TableCell className={props.classes.cell}>Region</TableCell>
      <TableCell className={props.classes.cell}>Reported Classification</TableCell>
      <TableCell className={props.classes.cell}>Last Evaluated</TableCell>
      <TableCell className={props.classes.cell}>Last Updated</TableCell>
      <TableCell className={props.classes.cell}>More Info</TableCell>
    </TableRow>
  </TableHead>
)

const styles = theme => ({
  row: {
      backgroundColor: '#3e5e9c',
  },
  cell: {
    color: 'white',
    fontSize: 20
  }
})


export default withStyles(styles)(VariantTableHeader)
