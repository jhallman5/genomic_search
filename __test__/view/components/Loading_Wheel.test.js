import React from 'react';
import { LoadingWheel } from '../../../src/view/components/Loading_Wheel';
import renderer from 'react-test-renderer';
import { MuiThemeProvider } from '@material-ui/core/'
import { shallow, mount } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  classes: {
    progress: {
      color: '#39B7AA',
      position: 'fixed',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)'
    }
  }
}
test('LoadingWheel renders', () => {
  const wrapper = shallow(<LoadingWheel {...props} />);
  const component = renderer.create(wrapper);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
