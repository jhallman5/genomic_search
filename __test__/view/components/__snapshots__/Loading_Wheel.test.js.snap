// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWheel renders 1`] = `
<div>
  <div
    aria-valuenow={0}
    className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 color position top left transform"
    role="progressbar"
    style={
      Object {
        "height": 75,
        "transform": "rotate(0.000deg)",
        "width": 75,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-6"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-7"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={
          Object {
            "strokeDasharray": "126.920",
            "strokeDashoffset": "126.920px",
          }
        }
      />
    </svg>
  </div>
</div>
`;
